volumina/tiling/__init__.py

```diff
@@ -552,6 +562,11 @@ def _fetch_layer_tile(self, timestamp, ims, transform, tile_nr, stack_id, ims_re
 
                 if stack_id == self._current_stack_id and cache is self._cache:
                     self.sceneRectChanged.emit(tile_rect)
+            else:
+                ims_req.cancel()
+
+            self._current_requests.pop((stack_id[1], ims, tile_nr), None)
```

Comment:
Are you using `pop` here  because the request might  have been already removed from `self._current_requests`?

This kinda makes me think if it would be possible for a request to overwrite an older one; If I, say, load an image in a layer, make a request, drop that layer, then load another image and make another request, is it possible for the second request to overwrite the previous one? Even worse, is it possible for the second one to be cancelled when we meant to cancel the first?